File: cpp/3090.cpp
#include <iostream>
#include <string>
#include <map>
using namespace std;

class Solution {
public:
    int maximumLengthSubstring(string s) {
        int m[26] = {0};
        int ans = 0;
        int j = 0;
        for (int i = 0; i < s.size(); ++i) {
            int ii = s[i] - 'a';
            ++m[ii];
            while (j < i && m[ii] > 2) {
                --m[s[j] - 'a'];
                ++j;
            }
            ans = max(ans, i - j + 1);
        }
        return ans;
    }
};

File: python3/3090.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        ans = 0
        m = {}
        j = 0
        for i in range(len(s)):
            if s[i] not in m:
                m[s[i]] = 1
            else:
                m[s[i]] += 1
            while j < i and m[s[i]] > 2:
                m[s[j]] -= 1
                j += 1
            ans = max(ans, i - j + 1)
        return ans